cpp/src/datetime/datetime_ops.cu

```diff
@@ -127,6 +127,23 @@ struct extract_day_num_of_year {
   }
 };
 
+// Extract the the quarter to which the timestamp belongs to
+struct extract_quarter_op {
+  template <typename Timestamp>
+  CUDA_DEVICE_CALLABLE int16_t operator()(Timestamp const ts) const
+  {
+    using namespace cuda::std::chrono;
+
+    // Only has the days - time component is chopped off, which is what we want
+    auto const days_since_epoch = floor<days>(ts);
+    auto const date             = year_month_day(days_since_epoch);
+    auto const month            = unsigned{date.month()};
+
+    return (month + 2) /
+           3;  // (x + y - 1) / y = ceil(x/y), where x and y are unsigned. x = month, y = 3
```

Comment:
Just for fun, you can look at the PTX generated by this new implementation vs. the old one and you can see how it gets rid of the branches in favor of a much more efficient add/multiply/shift: https://godbolt.org/z/Y5rW43bff
```suggestion
   // (x + y - 1) / y = ceil(x/y), where x and y are unsigned. x = month, y = 3
    return (month + 2) / 3;  
```

Reply:
Interesting!